### ISSUE TITLE
Compiler warnings

### ISSUE THREAD

**Issue by bgloyer** [MarcNo/lifelines#333]
This is to track some of the compiler warnings.

Iv'e been doing a little clean up in bgloyer/waringfixes.  A couple of the issues (see the TODOs) look like bugs.  It needs more clean-up before pulling.

**Comment by memmerto** [comment]
The bulk of what you have done in your fork has been handled in a more portable way in my warn_cleanup branch.  I will look at the TODOs that you have identified.

**Comment by memmerto** [comment]
Ignoring the %d/%ld changes which are non-portable, there are some good code cleanups in this branch.
(charmaps.c, dblist.c, indiseq.c, lex.c, browse.c, import.c, llexec.c, generic.c, lltest.c.)

If you could revert your %d/%ld changes, I would entertain a pull request for the remainer.

**Comment by bgloyer** [comment]
I reverted the d/ld changes and sent a pull request.  I see you are making fixing warnings in a branch too.  Will these conflict?  Did SonarCube give a lot of warnings?

**Comment by memmerto** [comment]
I've approved the pull request.  There is a ton of additional work going on in warn_cleanup (all the 64-bit compatibility stuff is happening there, so it's a lot more than warnings.  Hopefully that can be merged soon.)  I haven't been successful with SonarCube yet (the C/C++ support is paid license, which I only realized after getting everything set up.)  I have fixed a few things that cppcheck found though.
